src/runtime_src/core/tools/xbmgmt2/flash/xqspips.cpp

```diff
@@ -482,6 +482,24 @@ unsigned int XQSPIPS_Flasher::getFlashSize()
 
 void XQSPIPS_Flasher::readBack(const std::string& output, unsigned int base)
 {
+    initQSpiPS();
+
+    uint32_t StatusReg = XQSpiPS_GetStatusReg();
+
+    if (StatusReg == 0xFFFFFFFF) {
```

Comment:
This code is dangerous for it makes a lot of assumptions.  For example, what happens of StatusReg becomes a uint64_t.  It would be better to replace:
`0xFFFFFFFF`
with:
`std::numeric_limits<decltype(StatusReg)>::max()`

Reply:
ditto